docs/Chef-CLI-Catalog.md

```diff
@@ -0,0 +1,226 @@
+Go library. The design of this work will be captured in a separate document.
+
+### Route tools and binaries through the `chef` catalog
+
+The last thing we propose is routing all our tools and binaries through the `chef`
+catalog. An example would be the `hab` and `inspec` binaries that will be routed
+as `chef hab` and `chef inspec` accordingly.
+
+The complete output of the `chef` catalog should look like:
+```
+$ chef
+Usage:
+    chef -h/--help
+    chef -v/--version
+    chef command [arguments...] [options...]
+
+Available Commands:
+    env                     Prints environment variables used by Chef Workstation
+    gem                     Runs the 'gem' command in context of the embedded Ruby
+    exec                    Runs the command in context of the embedded ruby
+    shell-init              Initialize your shell to use Chef Workstation as your primary Ruby
+    generate                Generate a new repository, cookbook, or other component
+    describe-cookbook       Prints cookbook checksum information used for cookbook identifier
+    policy                  Manage Chef Policyfiles
+    hab                     Commands relating to Chef Habitat
+    inspec                  Tests and auditing your applications and infrastructure
+    kitchen                 Test cookbooks across any combination of platforms and test suites
+    knife                   An interface between a local chef-repo and the Chef Infra Server
+    ohai                    Collect system configuration data
+    client                  Agent that runs locally on every node that is under management by Chef Infra Server
+    run                     Execute ad-hoc tasks using Chef
+    apply                   Executable program that runs a single recipe
+    shell                   A recipe debugging tool that allows the use of breakpoints within recipes
+    berks                   Dependency manager for Chef cookbooks
+    cookstyle               Code linting tool that helps you write better Chef Infra cookbooks
+
+Relocated Commands:
+    install                 Use now: chef policy install
+    update                  Use now: chef policy update
+    push                    Use now: chef policy push
+    push-archive            Use now: chef policy push
+    show-policy             Use now: chef policy show
+    diff                    Use now: chef policy diff
+    export                  Use now: chef policy export
+    clean-policy-revisions  Use now: chef policy clean revisions
+    clean-policy-cookbooks  Use now: chef policy clean cookbooks
+    delete-policy-group     Use now: chef policy group delete
+    delete-policy           Use now: chef policy delete
+    undelete                Use now: chef policy undelete
+
+Experimental Commands:
+    analyze                 Analyze artifacts from a Chef Infra Server
+```
+
+### EXTRA: Present correct usage of sub-binaries
+
+An additional task from this proposal is the use of an environment variable that
```

Comment:
Talked with Salim about this - IMO, our first implementation should reference `chef policy` in all the help text. This makes it more confusing for our effortless developers who will directly invoke the `chef-policy` binary (without all the overhead of the entire Chef Workstation ecosystem). But those are internal users we can help.

In the future I could see us developing a Golang `effortless` executable that functions similarly to the `chef` golang executable. It allows us to start creating a custom workflow for them, leveraging tools under the covers that they do not care whether are written in Ruby or Golang. At that time I would introduce some kind of environment variable that would allow the help text to display `chef policy` or `effortless policy`.